File: Java/1491. Average Salary Excluding the Minimum and Maximum Salary.java
class Solution {
    public double average(int[] salary) {
        int total = 0, count = -2;
        int max = Integer.MIN_VALUE, min = Integer.MAX_VALUE;
        
        for (int n: salary) {
            total += n;
            count++;
            max = Math.max(max, n);
            min = Math.min(min, n);
        }
        
        return (double) (total-max-min) / count;
    }
}

File: Python3/1491. Average Salary Excluding the Minimum and Maximum Salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        total = count = 0
        minn = inf
        maxx = -inf
        
        for n in salary:
            total += n
            count += 1
            minn = min(minn, n)
            maxx = max(maxx, n)
            
        return (total - minn - maxx) / (count - 2)